File: src/libs/ThirdParty.VBSuppliers/Properties/Stuba/IStubaSettings.vb
Imports ThirdParty.Abstractions

Public Interface IStubaSettings
    ReadOnly Property URL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property MaxHotelsPerRequest(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
    ReadOnly Property Organisation(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Version(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Currency(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Nationality(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property ExcludeNonRefundableRates(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property ExcludeUnknownCancellationPolicys(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer
End Interface

File: src/libs/iVectorOne.Suppliers/Stuba/IStubaSettings.cs
namespace ThirdParty.CSSuppliers.Stuba
{

    public interface IStubaSettings
    {
        string GenericURL(IThirdPartyAttributeSearch tpAttributeSearch);
        int HotelBatchLimit(IThirdPartyAttributeSearch tpAttributeSearch);
        string Organisation(IThirdPartyAttributeSearch tpAttributeSearch);
        string User(IThirdPartyAttributeSearch tpAttributeSearch);
        string Password(IThirdPartyAttributeSearch tpAttributeSearch);
        string Version(IThirdPartyAttributeSearch tpAttributeSearch);
        string Currency(IThirdPartyAttributeSearch tpAttributeSearch);
        string LeadGuestNationality(IThirdPartyAttributeSearch tpAttributeSearch);
        bool ExcludeNRF(IThirdPartyAttributeSearch tpAttributeSearch);
        bool ExcludeUnknownCancellationPolicys(IThirdPartyAttributeSearch tpAttributeSearch);
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool IsMandatory);
    }
}